### PR TITLE
Fixes #3065 Broken error checking in OidcProviderConfiguration.parseURI

### LINKED ISSUE #3065: Broken error checking in OidcProviderConfiguration.parseURI?
While looking into https://github.com/couchbase/sync_gateway/issues/3025, I noticed this code in `oidc_provider.go`:

```
func (pc *OidcProviderConfiguration) parseURI(s string) (*url.URL, error) {

	if s == "" {
		return nil, nil
	}
	u, err := url.Parse(s)
	if err == nil {
		if u.Host == "" {
			return nil, fmt.Errorf("Host required in URI [%s]:", s)
		} else if u.Scheme != "http" && u.Scheme != "https" {
			return nil, fmt.Errorf("Invalid URI scheme [%s]:", s)
		}
	}
	return u, nil
}
```

Shouldn't that be `if err != nil` rather than `err == nil`?  As it stands, it looks like any error returned from `url.Parse()` will be ignored.

### PR DESCRIPTION
**Pull request by tleyden**
Fixes #3065

### CODE REVIEW

File: auth/oidc_provider.go [hunk 1]
```diff
@@ -132,12 +133,15 @@ func (pc *OidcProviderConfiguration) parseURI(s string) (*url.URL, error) {
...
 	}
 	u, err := url.Parse(s)
-	if err == nil {
-		if u.Host == "" {
-			return nil, fmt.Errorf("Host required in URI [%s]:", s)
-		} else if u.Scheme != "http" && u.Scheme != "https" {
-			return nil, fmt.Errorf("Invalid URI scheme [%s]:", s)
-		}
+	if err != nil {
+		return nil, pkgerrors.Wrapf(err, "Error parsing URI: [%s]", s)
```
Comment (adamcfraser) [inline auth/oidc_provider.go:137]: Previously this code was (incorrectly) not returning the error in this case.  Have we done any verification on whether making this fix breaks anything in our oidc support?

Comment (tleyden) [reply auth/oidc_provider.go:137]: I didn't do any verification, and I don't see any unit tests that cover this code (http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-coverage/16/artifact/cover_sg_merged.html#file3).

Comment (tleyden) [reply auth/oidc_provider.go:137]: I pushed a commit that adds a test that ensures that the common providers won't cause any regressions by returning URLs that don't parse, which could cause regressions due to the error checking change.

Comment (adamcfraser) [reply auth/oidc_provider.go:137]: That looks like a good approach - thanks.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**